addons/web/static/src/legacy/js/core/minimal_dom.js

```diff
@@ -101,10 +101,15 @@ export function makeButtonHandler(fct) {
         // part, the button is disabled without any visual effect.
         buttonEl.classList.add('pe-none');
         Promise.resolve(DEBOUNCE && new Promise(r => setTimeout(r, DEBOUNCE)))
-            .then(function () {
+            .then(async function () {
                 buttonEl.classList.remove('pe-none');
                 const restore = addButtonLoadingEffect(buttonEl);
-                return Promise.resolve(result).then(restore, restore);
+                try {
```

Comment:
how about just not return at the end ?
```js
Promise.resolve(result).finally(restore)
```
if result is already a promise, it will be the one on which the finally applies, since it is not returned by the then call back above the debounced Promise won't see it.
catch/resolve are handled by the caller at line 115.

sounds good ?

Reply:
It sounds good to me, but unfortunately not to the tests that I have added :slightly_frowning_face:
However, we could stick to good old `.then(restore, restore);`, would it suit you?